samsungtvws/art.py

```diff
@@ -121,7 +121,7 @@ def _send_art_request(
                 # Check sub event, reset event if it doesn't match
                 data = json.loads(response["data"])
                 sub_event = data.get("event", "*")
-                if sub_event != wait_for_sub_event:
+                if sub_event != wait_for_sub_event and sub_event != 'error':
```

Comment:
I think it would be cleaner to raise here straight away, and avoid RuntimeError. I think `ResponseError` would be more suitable, or you could create a custom error:
```suggestion
                if sub_event == "error":
                    raise exceptions.ResponseError(
                        f"`{request_data['request']}` request failed "
                        f"with error number {data['error_code']}"
                    )
                if sub_event != wait_for_sub_event:
```